pulsar-common/src/main/java/org/apache/pulsar/common/util/collections/TripleLongPriorityQueue.java

```diff
@@ -152,6 +174,21 @@ private void increaseCapacity() {
         buffer.capacity(this.capacity * TUPLE_SIZE);
     }
 
+    private void shrinkCapacity() {
+        if (capacity > initialCapacity &&  size < capacity * shrinkFactor) {
```

Comment:
We could cache the value of `capacity * shrinkFactor` to avoid computing it each time we're removing an item